spec/io_spec.rb

```diff
@@ -18,9 +21,15 @@
 
   describe 'get_input' do
     it 'receives input from the user' do
-      io.get_input
+      expect(io.get_input('message', method(:callback))).to eql('input')
+      expect(input.check_message).to eql('message')
       expect(input.times_called).to eql(1)
-      expect(input.gets).to eql('input')
+    end
+
+    it 'exits the game if EOFError passes nil' do
+      input.set_input(nil)
+      expect(io).to receive(:exit)
```

Comment:
How can we refactor this so that we are not using the Rspec mock API?

Reply:
Changed this so a call to exit doesn't appear.  Passes a symbol around now, which will have meaning further up.